core/src/main/java/io/questdb/griffin/engine/join/JoinRecordMetadata.java

```diff
@@ -183,7 +183,7 @@ private int addAlias(CharSequence tableAlias, CharSequence columnName) {
 
         MapValue value = key.createValue();
         if (!value.isNew()) {
-            throw CairoException.instance(0).put("Duplicate column [name=").put(columnName).put(", tableAlias=").put(tableAlias).put(']');
+            throw CairoException.duplicateColumn(columnName, tableAlias);
```

Comment:
There might be another bug here. This exception is not tested with real factories. Throwing it when `join` is being assembled can lead to factories not being freed and ultimately memory leak.

It would be good to have a test to ensure resources are freed when this exception is thrown in the correct context.

Reply:
I will proceed to test it.

Reply:
@bluestreak01 I don't think I can come up with a join statement that will result in throwing the exception. I have created test `AsOfJoinTest.testLtJoinAttemptToBlowUpJoinRecordMetadata` to exemplify what I mean. Any ideas are welcome :p